gemini/evaluation/bring_your_own_computation_based_metric.ipynb

```diff
@@ -0,0 +1,1787 @@
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "en_HfAkdtcRn"
+   },
+   "source": [
+    "### Set Google Cloud project information and initialize Vertex AI SDK for Python\n",
+    "\n",
+    "To get started using Vertex AI, you must have an existing Google Cloud project and [enable the Vertex AI API](https://console.cloud.google.com/flows/enableapi?apiid=aiplatform.googleapis.com). Learn more about [setting up a project and a development environment](https://cloud.google.com/vertex-ai/docs/start/cloud-environment)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "id": "V8M4F84GthAr"
+   },
+   "outputs": [],
+   "source": [
+    "PROJECT_ID = \"[your-project-id]\"  # @param {type:\"string\"}\n",
+    "LOCATION = \"us-central1\"  # @param {type:\"string\"}\n",
+    "EXPERIMENT_NAME = \"customize-metrics\"  # @param {type:\"string\"}\n",
+    "\n",
+    "if not PROJECT_ID or PROJECT_ID == \"[your-project-id]\":\n",
+    "    raise ValueError(\"Please set your PROJECT_ID\")\n",
+    "\n",
+    "\n",
+    "import vertexai\n",
+    "\n",
+    "vertexai.init(project=PROJECT_ID, location=LOCATION)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "EWTezEusurJU"
+   },
+   "source": [
+    "### Import libraries\n",
+    "\n",
+    "Import the Vertex AI Python SDK and other required Python libraries."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "id": "cpoVotOuzqsb"
+   },
+   "outputs": [],
+   "source": [
+    "# General\n",
+    "from IPython.display import display, Markdown, HTML\n",
+    "import inspect\n",
+    "import plotly.graph_objects as go\n",
+    "\n",
+    "\n",
+    "# Main\n",
+    "import pandas as pd\n",
+    "from vertexai.generative_models import GenerativeModel\n",
+    "from vertexai.evaluation import (\n",
+    "    EvalTask,\n",
+    "    CustomMetric,\n",
+    "    MetricPromptTemplateExamples,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "42svczJqjZOw"
+   },
+   "source": [
+    "### Helper functions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "id": "bLbaS-67jZOw"
+   },
+   "outputs": [],
+   "source": [
+    "def print_doc(function):\n",
+    "    print(f\"{function.__name__}:\\n{inspect.getdoc(function)}\\n\")\n",
+    "\n",
+    "\n",
+    "def display_eval_result(eval_result, title=None, metrics=None):\n",
+    "    \"\"\"Display the evaluation results.\"\"\"\n",
+    "    summary_metrics, metrics_table = (\n",
+    "        eval_result.summary_metrics,\n",
+    "        eval_result.metrics_table,\n",
+    "    )\n",
+    "\n",
+    "    metrics_df = pd.DataFrame.from_dict(summary_metrics, orient=\"index\").T\n",
+    "    if metrics:\n",
+    "        metrics_df = metrics_df.filter(\n",
+    "            [\n",
+    "                metric\n",
+    "                for metric in metrics_df.columns\n",
```

Comment:
These types of display functions seem to be used a lot in the notebooks. I'm wondering if it could make sense to add them directly to the SDK?

Reply:
Those are Jupyter notebook specific IPython.display helper functions. It might not apply to all SDK users.